cgi-bin/LJ/Web.pm

```diff
@@ -777,6 +777,54 @@ sub bad_password_redirect {
     }
 }
 
+# <LJFUNC>
+# name: LJ::icons_for_remote
+# class: web
+# des: Gets all the userpics for a given user, including an item for the default userpic.
+# args: remote
+# des-remote: a user object.
+# returns: An array of {value => ..., text => ..., data => { url => ...}}
+#          hashrefs, suitable for use as the items in an LJ::html_select().
+#          If no userpics or if remote is undefined, an empty array.
+# </LJFUNC>
+sub icons_for_remote {
+    my ($remote) = @_;
+    my @pics;
+
+    {
+        my %res;
+        if ($remote) {
+            LJ::do_request(
```

Comment:
Just a note for future spelunkers... this is such a weird design to call the protocol handler and "log in" to get the userpics.

Reply:
man, I was like, I don't even want to figure out what's happening in here, just isolate the infection and g e t  o u t